sources/logger.c

```diff
@@ -32,7 +32,52 @@ static int od_log_syslog_level[] = { LOG_INFO, LOG_ERR, LOG_DEBUG, LOG_CRIT };
 
 static char *od_log_level[] = { "info", "error", "debug", "fatal" };
 
-void od_logger_init(od_logger_t *logger, od_pid_t *pid)
+static inline void _od_logger_write(od_logger_t *l, char *data, int len,
+				    od_logger_level_t lvl)
+{
+	int rc;
+	if (l->fd != -1) {
+		rc = write(l->fd, data, len);
+	}
+	if (l->log_stdout) {
+		rc = write(STDOUT_FILENO, data, len);
+	}
+	if (l->log_syslog) {
+		syslog(od_log_syslog_level[lvl], "%.*s", len, data);
+	}
+	(void)rc;
+}
+
+static inline void od_logger(void *arg)
+{
+	od_logger_t *logger = arg;
+
+	for (;;) {
+		machine_msg_t *msg;
+		msg = machine_channel_read(logger->task_channel, UINT32_MAX);
```

Comment:
How are we going to shutdown? When Odyssey get SEGQUIT or SIGTERM?

Reply:
yes, just like other threads